File: app/src/main/java/com/example/walmartpathfinder/models/ShoppingList.kt
package com.example.walmartpathfinder.models

class ShoppingList(
    private var title: String,
    private var shoppingList: MutableList<Item> = mutableListOf(),
) {
    /* Add Items to List */
    fun addItem(item: Item) {
        shoppingList.add(item)
    }

    /* Remove Items from List */
    fun removeItem(item: Item) {
        if (shoppingList.contains(item)) {
            shoppingList.remove(item)
        }
    }

    override fun toString(): String {
        return this.title
    }

}

File: src/models/ShoppingList.ts
import { Item } from "./Item";

class ShoppingList{
    shoppingList: Item[] = []

    addItem(item: Item){
        this.shoppingList.push(item)
    }

    removeItem(item: Item){
        this.shoppingList.filter((val => {
            return val != item;
        }))
    }
}